pgx/src/bgworkers.rs

```diff
@@ -73,6 +73,7 @@ impl BackgroundWorker {
         const LEN: usize = 96;
 
         unsafe {
+            assert!(!pg_sys::MyBgworkerEntry.is_null(), "BackgroundWorker associated functions can only be called from a registered background worker");
```

Comment:
I think these should be `debug_assert`s to avoid a perf impact in release?

Reply:
I agree that it is unlikely that a mistake such as invoking background-worker methods gets into a release build, as it should manifest pretty quickly in development.

On the other hand, I believe only `BackgroundWorker::transaction()` has the potential impact performance in any meaningful way, as others are unlikely to be called in any kind of tight(ish) loop.

Reply:
Dereferencing the null pointer is unsound. If it is at all in question whether it is null, it should be asserted on or the function should return a sum type. These functions return bare types, so they **must** assert.